src/ImageCore.jl

```diff
@@ -33,13 +33,17 @@ const RRArray{To,From,N,M,P} = Base.ReshapedArray{To,N,Base.ReinterpretArray{To,
 const RGArray = Union{Base.ReinterpretArray{<:AbstractGray,M,<:Number,P}, Base.ReinterpretArray{<:Number,M,<:AbstractGray,P}} where {M,P}
 
 # delibrately not export these constants to enable extensibility for downstream packages
-const NumberLike{T<:Number} = Union{T,AbstractGray{T}}
+const PixelLike{T<:Number} = Union{T, Colorant{T}}
+const NumberLike{T<:Number} = Union{T, AbstractGray{T}}
 const RealLike{T<:Real} = NumberLike{T}
 const FloatLike{T<:AbstractFloat} = RealLike{T}
 const FractionalLike{T<:Union{FixedPoint, AbstractFloat}} = RealLike{T}
 const GrayLike{T<:Union{Bool, FixedPoint, AbstractFloat}} = RealLike{T}
-const GenericGrayImage{N, T<:GrayLike} = AbstractArray{<:GrayLike{T}, N}
-const Gray2dImage{T<:GrayLike} = GenericGrayImage{2, T}
+const GenericImage{N, T<:Number} = AbstractArray{<:PixelLike{T}, N}
```

Comment:
I worry it will be hard to remember that `T` is the "inner" eltype rather than the "outer" eltype. Is the RHS really that bad for dispatch? I.e.,

```julia
foo(A::AbstractArray{<:PixelLike})
```

seems rather easy to me.